beacon-chain/powchain/service_test.go

```diff
@@ -264,6 +264,51 @@ func TestStop_OK(t *testing.T) {
 	hook.Reset()
 }
 
+func TestRPCRestart_OK(t *testing.T) {
+	hook := logTest.NewGlobal()
+
+	endpoint := "ws://127.0.0.1"
+	testAcc, err := setup()
+	if err != nil {
+		t.Fatalf("Unable to set up simulated backend %v", err)
+	}
+	gr := &goodReader{}
+	web3Service, err := NewWeb3Service(context.Background(), &Web3ServiceConfig{
+		Endpoint:        endpoint,
+		DepositContract: testAcc.contractAddr,
+		Reader:          gr,
+		Logger:          &goodLogger{},
+		BlockFetcher:    &goodFetcher{},
+		ContractBackend: testAcc.backend,
+	})
+	if err != nil {
+		t.Fatalf("unable to setup web3 ETH1.0 chain service: %v", err)
+	}
+	testAcc.backend.Commit()
+
+	web3Service.Start()
+
+	msg := hook.LastEntry().Message
+	want := "Could not connect to ETH1.0 chain RPC client"
+	if strings.Contains(want, msg) {
+		t.Errorf("incorrect log, expected %s, got %s", want, msg)
+	}
+	// clientPTR := &web3Service.client
+	// web3Service.headerChan <-
+	// time.Sleep(5 * time.Second)
+	// msg = hook.LastEntry().Message
+	// want = "EOF"
+	// if msg != want {
+	// 	t.Errorf("incorrect log, expected %s, got %s", want, msg)
+	// }
+	// newClientPTR := &web3Service.client
+	// if clientPTR == newClientPTR {
+	// 	t.Errorf("client wasn't reinitialized after EOF")
+	// }
```

Comment:
What's up with this commented code?

Reply:
couldn't make it send eof error to the header subscription

Reply:
you can't do `gr.feed.Send(io.EOF)`?